### PR TITLE
allow loading pretained Inception

### PR DESCRIPTION
**Pull request by shaform**
Sometimes one might want to finetune a Inception model pretrained on Imagenet. Such model does not contain the `Logits` fully-connected layer. Therefore, we must specify `ignore_missing_vars`. `checkpoint_exclude_scopes` could also be used to only load certain layers.

### CODE REVIEW

**Comment by shaform** [comment]
close due to inactivity

File: src/train_softmax.py [hunk 1]
```diff
@@ -205,7 +205,16 @@ def main(args):
...
+                if args.checkpoint_exclude_scopes:
+                    exclusions = [args.checkpoint_exclude_scopes.split(',')]
+                else:
+                    exclusions = []
+                # load from pre-trained model
+                tf.contrib.framework.assign_from_checkpoint_fn(
+                    pretrained_model,
+                    [var for var in tf.trainable_variables()
+                        if all(not var.op.name.startswith(exclusion) for exclusion in exclusions)],
+                    ignore_missing_vars=args.ignore_missing_vars)(sess)
```
Comment (sadicLiu) [inline src/train_softmax.py]: Hi, I'm confused of this piece of code, why follows(sess) after invoke tf.contrib.framework.assign_from_checkpoint_fn?

**Comment by wencoast** [comment]
@shaform @davidsandberg  I read some papers which are published recently. For example, [CosFace](https://arxiv.org/abs/1801.09414), [SphereFace](https://arxiv.org/abs/1704.08063), [ArcFace](https://arxiv.org/abs/1801.07698). 

All of them are training the model from scratch using the CASIA-WebFace dataset. 
Why don't they use the ImageNet weights to fine-tune the network model? 

@davidsandberg why don't you use the ImageNet weights to initialize the model weights?
